### ISSUE TITLE
Rework platform manifest

### ISSUE THREAD

**Issue by MSe1969** [cm12-amami/discussion#4]
Hi @hahnjo @Micha-Btz 

my idea of a reworked manifest outlined:

1. **purge file qrd.xml**
it is not referenced at all in default.xml and does not seem to have any meaning for the synch process. The CM commit history also only shows one commit of the file having been created and a 2nd one to replace cm by Lineage

2. **New file structure**
A structure consisting of three files (ONLY one branch: 'cm-12.1'):
- default.xml
- aosp.xml
- lineageos.xml

2.1 **aosp.xml** is referenced by an include statement from default.xml and contains all active AOSP links for tag **android-5.1.1_r38**. Instead of a remove-project tag in the local manifest, replaced lines will be removed.

2.2 **lineageos.xml** is referenced by an include statement from default.xml and contains all active LineageOS links for branch **cm-12.1**. Instead of a remove-project tag in the local manifest, replaced lines will be removed from here.

2.3 **default.xml** will contain three sections:
- a 'header' section defining the defaults and include statements to aosp.xml and lineageos.xml
- a 'cm12-amami' secition, where (nearly) all cm12-amami repositories are linked
- a 'special' section, where further repositories (e.g. privacy browser) and links to different AOSP tags (e.g. android-6.0.1_r81) or LineageOS branches (e.g. cm-13.0, cm-14.1) are listed

3. **Local Manifest**
We use the three branches cm-12.1, cm-12.1-test, cm-12.1-twrp, as before

3.1 *branch 'cm-12.1'*
The local manifest will only contain the device-specific repositories of cm12-amami.
It is expected that this manifest will almost never be edited any more.

3.2 *branch 'cm-12.1-test'*
Here, for testing purposes, we will replace either LinageOS/cm-12.1 or AOSP branches with cm12-amami/cm-12.1-test or even cm12-amami/cm-12.1 with cm-12.1-test.
Once we are ready to ship, we will update, if necessary the platform manifest and merge the repos, if we only have cm12-amami test branches, after merging, we'll delete the replacement lines.

3.3 *branch 'cm-12.1-twrp'*
Accordingly, this local manifest will be adapted following above logic. If we have time, we could work on simply integrating TWRP into the regular build process, same as OmniROM does...

Comments?

**Comment by Micha-Btz** [comment]
I didn't see much benefit from it. To include twrp in the build makes sense to me.

**Comment by MSe1969** [comment]
Have done now the re-design as described - test build is currently in process . . .

**Comment by MSe1969** [comment]
Three small additions to the documentation above (you may already have noticed):
1. In the *local manifest* for cm-12.1-test, I have created a 2nd file testing.xml, where all replacements during test phase should go in, so manifest.xml should 'correspond' to the one in cm-12.1 - makes it easier to merge the test branch later
2. *Main manifest* - the to-be-included manifests are in a subfolder
3. *Main manifest* - a 3rd include-manifest containing the Sony-stuff not specific to **our** device

Once you confirm it also working for you, I think we can close this issue. Please advise

**Comment by MSe1969** [comment]
Closing issue - several test build have been created successfully. For the TWRP build included into main build, we should create an own issue

**Comment by Micha-Btz** [comment]
I did some tests about adding twrp to regular build. This will require a lot of work with cm-12.1 so that in my opinion this is more work than necessary. For example a complete overhaul of the https://github.com/cm12-amami/android_device_sony_msm8974-common/blob/cm-12.1/boot/custombootimg.mk
So we should leave it on the state that it has at the moment and build twrp updates with the different manifest..
By the way I successfully build twrp 3.2.0 based on the oreo sources. I will upload this the next few days after I did some more tests.
